### PR TITLE
Add overview simulator

### PR DESCRIPTION
**Pull request by BrendanAnnable**
Direct port from NUsight1, minimal thought went into refactoring.

I've also just realised that a simulator no longer has control over its own frequency, so this would be sending 60 overview messages/second.

Probably needs some reworking. Handing this code and PR over to you @MonicaOlejniczak 🙂 Adapt as you need.

### CODE REVIEW

File: src/simulators/overview_simulator.ts [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      ballPosition,
+      ballWorldPosition,
+      gameMode: Mode.NORMAL,
+      gamePhase: Phase.INITIAL,
+      penaltyReason: PenaltyReason.UNPENALISED,
+      lastCameraImage: { seconds: time - 5000 * this.random() },
+      lastSeenBall: { seconds: time - 5000 * this.random() },
+      lastSeenGoal: { seconds: time - 15000 * this.random() },
+      lastSeenObstacle: { seconds: time - 15000 * this.random() },
+      pathPlan: [
```
Comment (BrendanAnnable) [inline src/simulators/overview_simulator.ts]: Is this no longer a thing @TrentHouliston?

File: src/simulators/overview_simulator.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        robotPosition,
+        this.randomFieldPosition(),
+        this.randomFieldPosition(),
+        this.randomFieldPosition(),
+        ballWorldPosition
+      ],
+      kickTarget: this.randomFieldPosition(),
```
Comment (BrendanAnnable) [inline src/simulators/overview_simulator.ts]: Is this still a thing? @TrentHouliston

File: src/server/prod.ts [hunk 3]
```diff
@@ -35,6 +38,7 @@ if (withSimulators) {
     numRobots: 1,
     simulators: [
       SensorDataSimulator.of(),
+      OverviewSimulator.of(),
     ],
   })
   virtualRobots.simulateWithFrequency(60)
```
Comment (BrendanAnnable) [inline src/server/prod.ts]: - [ ] TODO: I'll have to wrap each simulator separately.

File: src/simulators/overview_simulator.ts [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      robotHeading,
+      ballPosition,
+      ballWorldPosition,
+      gameMode: Mode.NORMAL,
+      gamePhase: Phase.INITIAL,
+      penaltyReason: PenaltyReason.UNPENALISED,
+      lastCameraImage: { seconds: time - 5000 * this.random() },
+      lastSeenBall: { seconds: time - 5000 * this.random() },
+      lastSeenGoal: { seconds: time - 15000 * this.random() },
+      lastSeenObstacle: { seconds: time - 15000 * this.random() },
```
Comment (BrendanAnnable) [inline src/simulators/overview_simulator.ts]: Do we detect obstacles?

Comment (TrentHouliston) [inline src/simulators/overview_simulator.ts]: We do not detect obstacles at the moment

File: src/simulators/overview_simulator.ts (hunk 1 above)
Comment (TrentHouliston) [inline src/simulators/overview_simulator.ts]: The path plan doesn't exist in that form any more no. But we do have a ball approach position point that we could use.

File: src/simulators/overview_simulator.ts (hunk 2 above)
Comment (TrentHouliston) [inline src/simulators/overview_simulator.ts]: We still have a kick target

**Pull request merged**